### PR TITLE
Added Reply:To for email request through Mailgun

### PR DESCRIPTION
**Pull request by ksmandersen**
I needed to add a reply-to for certain mails in my backend. 

I don't know if makes sense to use the extendedFields of Email to pass the reply-to address along. The extended fields only provide Strings while the reply-to should really be an array of addresses like from and to.

I followed the Mailgun API docs for adding reply-to from here:
http://mailgun-documentation.readthedocs.io/en/latest/api-sending.html#sending

Please let me know if I can approve this in any way.

### CODE REVIEW

File: Sources/Vapor/Mail/Mailgun.swift [hunk 1]
```diff
@@ -90,6 +90,14 @@ public final class Mailgun: MailProtocol {
             bodyKey: body
         ]
         
+        if let replyTo = mail.extendedFields["h:Reply-To"] {
+            let part = Part(headers: [:], body: replyTo.makeBytes())
+            req.formData?["h:Reply-To"] = Field(name: "h:Reply-To",
+                                                filename: nil,
```
Comment (tanner0101) [inline Sources/Vapor/Mail/Mailgun.swift]: no xcode formatting please :)

**Comment by tanner0101** [comment]
Should we also add an extension on `SMTP.Email` for setting the reply to address or does this already exist?

File: Sources/Vapor/Mail/Mailgun.swift (hunk 1 above)
Comment (ksmandersen) [inline Sources/Vapor/Mail/Mailgun.swift]: Not sure what you mean by Xcode formatting?

**Comment by ksmandersen** [comment]
@tanner0101 I updated the formatting. Anything else I can do to move this forward?

File: Sources/Vapor/Mail/Mailgun.swift [hunk 2]
```diff
@@ -90,6 +90,11 @@ public final class Mailgun: MailProtocol {
             bodyKey: body
         ]
         
+        if let replyTo = mail.extendedFields["h:Reply-To"] {
```
Comment (tanner0101) [inline Sources/Vapor/Mail/Mailgun.swift]: Should we make this more general and simple add all extended fields to the form data?

File: Sources/Vapor/Mail/Mailgun.swift (hunk 1 above)
Comment (tanner0101) [inline Sources/Vapor/Mail/Mailgun.swift]: Sorry for missing this. Xcode indents the parameter labels to the right of the function's opening `(`. This looks nice in Xcode but is really hard to recreate if you're using something like Sublime Text (which some server-side devs use). 

i.e., this:
```
Field(name: "h:Reply-To",
         filename: nil,
```

**Pull request merged**